02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,355 @@
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping zebrafish Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the first column of `df`
+head(df[, 1])
+```
+
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG`.
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_symbol_df <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol) %>%
+  # only retain the genes from the expression matrix that are in zebrafish data
+  dplyr::inner_join(df, by = c("zebrafish_ensembl" = "Gene"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_symbol_df, 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of symbols in the `Symbol` column
+summary(as.factor(human_symbol_df$human_symbol)) # We need to use the `as.factor()` function here in order to get the count of each unique value returned
+```
+
+Looks like we have mapped symbols!
+It does not explicitly tell us if we have IDs that did not map to symbols, which would result in NAs so let's explore to find out if we do have any.
+
+```{r}
+any(is.na(human_symbol_df$human_symbol)) # Logical statement to check if there are any NAs in the `human_symbol` column of our mapped data frame
+```
+
+There are no NAs!
+This means that all of our zebrafish Ensembl IDs mapped to at least one human gene symbol!
+This is exciting because it is not always the case everything maps across species and that is okay.
+
+However, if you have almost all NAs it is possible that the function was executed incorrectly or you may want to consider using a different gene identifier, if possible.
+
+## Write results to file
+
+Now let's write our results to file.
+
+```{r}
+readr::write_tsv(
+  human_symbol_df,
+  file.path(
+    results_dir,
+    "GSE71270_ensembl_to_human_gene_symbol.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+*Some things to note:*
+
+* Multiple zebrafish Ensembl gene IDs map to the same human symbol which results in duplicated human gene symbols.
+Some downstream tools may need this to be resolved.
+One way to resolve this may be to map to a different gene identifier.
```

Comment:
I know you pointed this out in your PR summary, but can you say more about what you mean by this?

Reply:
In other words, another gene identifier may have a better 1:1 ratio to the zebrafish Ensembl IDs, i.e. human Ensembl IDs instead of symbols (like I see you mention below).

Reply:
I would use the wording here instead of what you have in the notebook. And probably move away from using the word `resolve` in this context since it's not so much a problem to be solved as it is just the nature of how IDs map across species and between ID types (as in ortholog mapping is shaky at best).